server/data/chat.go

```diff
@@ -0,0 +1,126 @@
+package data
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/gorilla/websocket"
+	"github.com/jackc/pgx/v5"
+)
+
+type ChatMessage struct {
+	Type    string `json:"type,omitempty"`
+	From    string `db:"from_user" json:"from,omitempty"`
+	To      string `db:"to_user" json:"to"`
+	Content string `db:"content" json:"content"`
+}
+
+type ChatMessages struct {
+	Type          string         `json:"type"`
+	Conversations []*ChatMessage `json:"conversations"`
+}
+
+func JoinChannel(conn *websocket.Conn, username string) {
+	sub := redisClient.Subscribe(context.Background(), username)
+	defer sub.Close()
+
+	go loadMessages(conn, username)
+
+	chatMessage := ChatMessage{}
+	for {
+		// Listen for new messages
+		msg, err := sub.ReceiveMessage(context.Background())
+		if err != nil {
+			fmt.Println(err.Error())
```

Comment:
nitpick: Println takes in an `error` and calls `error.Error` for you, so no need to call it yourself.